packages/qgrid-ngx-plugins/src/lib/data-provider/cache-requested-pages.ts

```diff
@@ -0,0 +1,45 @@
+import { GridModel } from '@qgrid/ngx';
+import { Observable, of } from 'rxjs';
+import { filter, tap } from 'rxjs/operators';
+import { DataProviderPageServer } from './data-provider-page-server';
+import { DataProviderProcessContext } from './data-provider-process-context';
+import { DataProviderStrategy } from './data-provider-strategy';
+
+export class CacheAlreadyRequestedPageStrategy<T> implements DataProviderStrategy<T> {
+	private gridRowsCache: Map<number, T[]> = new Map();
+
+	constructor(
+		private server: Pick<DataProviderPageServer<T>, 'getPage'>,
+		private pagesToLoad: number = 0,
```

Comment:
it should be options `{ pagesToLoad: number }`, consider to rename pagesToLoad to numberOfBackgroundPages

Reply:
done

Reply:
done